salsah/src/public/js/jquery.searchbox.js

```diff
@@ -217,6 +224,15 @@
 								$this.css({cursor: 'text'});
 							}
 							localdata.ele.sel.hide();
+							// when leaving this input, if the value is empty, but we had a previous valid input
+							// set back the previous input
+							if ($this.attr("isValid") && $this.attr("isValid") == "false") {
```

Comment:
I simply try to avoid relying on type conversions because this could have unexpected effects. Whenever possibly, I use `===` or `!==` (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Comparison_Operators).